ingestion/src/metadata/ingestion/source/database/dbt/dbt_service.py

```diff
@@ -169,51 +173,28 @@ def remove_manifest_non_required_keys(self, manifest_dict: dict):
             }
         )
 
-        required_nodes_keys = {
-            "schema_",
-            "schema",
-            "name",
-            "resource_type",
-            "path",
-            "unique_id",
-            "fqn",
-            "alias",
-            "checksum",
-            "config",
-            "column_name",
-            "test_metadata",
-            "original_file_path",
-            "root_path",
-            "database",
-            "tags",
-            "description",
-            "columns",
-            "meta",
-            "owner",
-            "created_at",
-            "group",
-            "sources",
-            "compiled",
-            "docs",
-            "version",
-            "latest_version",
-            "package_name",
-            "depends_on",
-            "compiled_code",
-            "compiled_sql",
-            "raw_code",
-            "raw_sql",
-            "language",
-        }
+        for field in ["nodes", "sources"]:
+            for node, value in manifest_dict.get(  # pylint: disable=unused-variable
+                field
+            ).items():
```

Comment:
@SumanMaharana here we should probably check `manifest_dict.get(...` returns a dict and is not Null -- we can also update the method signature `def remove_manifest_non_required_keys(self, manifest_dict: dict):` if we are expecting indeed a `Dict[str, dict]`.


Otherwise we'll get 
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'str' object has no attribute 'items'
```
or 
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'items'
```

Reply:
@TeddyCr `manifest_dict.get(...` will always return a dict as its defined that way in dbt.